wgpu-hal/src/dx12/command.rs

```diff
@@ -380,6 +379,101 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
         }
     }
 
+    unsafe fn clear_texture(
+        &mut self,
+        texture: &super::Texture,
+        subresource_range: &wgt::ImageSubresourceRange,
+    ) {
+        // Note that CopyTextureRegion for depth/stencil or multisample resources would require full subresource copies.
+        // Meaning we'd need a much larger pre-zeroed buffer
+        // (but instead we just define clear_texture to not support these)
+
+        let list = self.list.unwrap();
+        let mut src_location = d3d12::D3D12_TEXTURE_COPY_LOCATION {
+            pResource: self.shared.zero_buffer.as_mut_ptr(),
+            Type: d3d12::D3D12_TEXTURE_COPY_TYPE_PLACED_FOOTPRINT,
+            u: mem::zeroed(),
+        };
+        let mut dst_location = d3d12::D3D12_TEXTURE_COPY_LOCATION {
+            pResource: texture.resource.as_mut_ptr(),
+            Type: d3d12::D3D12_TEXTURE_COPY_TYPE_SUBRESOURCE_INDEX,
+            u: mem::zeroed(),
+        };
+        let raw_format = conv::map_texture_format(texture.format);
+        let format_desc = texture.format.describe();
+
+        let depth = if texture.dimension == wgt::TextureDimension::D3 {
+            texture.size.depth_or_array_layers
+        } else {
+            1
+        };
+
+        let mip_range = subresource_range.base_mip_level
+            ..(subresource_range.base_mip_level
+                + subresource_range
+                    .mip_level_count
+                    .map_or(texture.mip_level_count, |c| c.get()));
```

Comment:
this is incorrect, it should be something like
```rust
match subresource_range.mip_level_count {
  Some(c) => subresource_range.base_mip_level + c.get(),
  None => texture.mip_level_count,
}
```

Otherwise, the non-zero `base_mip_level` with no `mip_level_count` would lead to an invalid range